crates/uv-workspace/src/workspace.rs

```diff
@@ -746,9 +764,34 @@ impl Workspace {
             .and_then(|uv| uv.index)
             .unwrap_or_default();
 
+        let required_members = workspace_sources
```

Comment:
This isn't strictly correct, as you could have an unused source here? However, if the workspace member is a dependency, you _do_ need the source entry. I'm hesitant to add collection of the full requirement graph here. We might need to consider a different approach.

Unfortunately, I think we need to do something like this (collect the required members when constructing the `Workspace`) because in places like `Workspace:: members_requirements` we don't otherwise have enough information to construct the proper `editable` and `virtual` values. It was "working" before #14197, but somehow that leads to conflicting URL errors now.

Reply:
(You can see the previous approach prior to https://github.com/astral-sh/uv/pull/14663/commits/94efd22fcf3cd64ae02d2a334f1ad37d8c4a476c)

Reply:
cc @konstin in case you have any ideas here

We could also just not differentiate workspace member build behavior based on whether or not its a dependency, that'd make life way easier.